src/mixer/playermanager.cpp

```diff
@@ -361,67 +358,73 @@ void PlayerManager::addConfiguredDecks() {
 
 void PlayerManager::addDeckInner() {
     // Do not lock m_mutex here.
-    QString group = groupForDeck(m_decks.count());
-    VERIFY_OR_DEBUG_ASSERT(!m_players.contains(group)) {
+    ChannelHandleAndGroup handle_group =
+            m_pEngine->registerChannelGroup(groupForDeck(m_decks.count()));
```

Comment:
This feels hacky. It seems EngineMaster::registerChannelGroup is public only for the tests. I think using `friend class` would be better for that. For here, let's pass the ChannelHandleFactory to PlayerManager.

Reply:
Actually, it is used in a few more places. As I said below, it is not really "register"ing anything, so maybe it would be enough to rename it?